cmd/promtool/sd.go

```diff
@@ -0,0 +1,143 @@
+// Copyright 2021 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/go-kit/log"
+
+	"github.com/prometheus/prometheus/config"
+	"github.com/prometheus/prometheus/discovery"
+	"github.com/prometheus/prometheus/discovery/targetgroup"
+	"github.com/prometheus/prometheus/pkg/labels"
+	"github.com/prometheus/prometheus/pkg/relabel"
+)
+
+type sdCheckResult struct {
+	Target          string
+	OriginalLabels  labels.Labels
+	AfterRelabeling labels.Labels
+}
+
+// CheckSD performs service discovery for the given job name and reports the results.
+func CheckSD(sdConfigFiles, sdJobName string, sdTimeout time.Duration) int {
```

Comment:
This is looking good, aside from the few comments below. However it needs testing. I think it would make sense that this function takes the place to output to, i.e. `os.Stdout` in normal running. Then write a test that provides its own buffer and at least checks the JSON is valid.